src/main/java/uk/gov/register/store/postgres/BatchedPostgresDataAccessLayer.java

```diff
@@ -0,0 +1,351 @@
+package uk.gov.register.store.postgres;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import uk.gov.register.core.*;
+import uk.gov.register.db.*;
+import uk.gov.register.store.DataAccessLayer;
+import uk.gov.register.util.HashValue;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class BatchedPostgresDataAccessLayer implements DataAccessLayer {
+    private final Map<EntryType, TreeMap<Integer, Entry>> batchedEntries;
+    private final Map<EntryType, Map<String, Integer>> batchedRecords;
+    private final Map<HashValue, Item> batchedItems;
+
+    private final PostgresDataAccessLayer postgresDataAccessLayer;
+
+    private boolean isEntireRegisterBatched;
+
+    public BatchedPostgresDataAccessLayer(PostgresDataAccessLayer postgresDataAccessLayer) {
```

Comment:
The short answer is, _I'm not sure_ but I think we should make a separate task to review the batching implementation rather than doing it on this branch unless functionality is impaired